### PR TITLE
added monday handler

### PR DESCRIPTION
**Pull request by alon710**

### CODE REVIEW

File: handlers/monday.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
+from monday import MondayClient
```
Comment (alon710) [inline handlers/monday.py]: Run pre-commit

File: scraper.py [hunk 2]
```diff
@@ -29,6 +29,13 @@ def get_logger() -> logging.Logger:
     username = config["MOODLE_USERNAME"]
     password = config["MOODLE_PASSWORD"]
     base_url = config["MOODLE_BASEURL"]
+    monday_token = config["MONDAY_API_KEY"]
+    monday = MondayHandler(monday_token)
+    boards = monday.get_boards()
+    board = boards.fetch_boards()
+    board_id = 5818273381
```
Comment (alon710) [inline scraper.py]: Should be an environment variable also

File: handlers/monday.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+import requests
+import os
+from handlers.moodle import MoodleHandler
+class MondayHandler:
+    def __init__(self, token):
+        self.client = MondayClient(token)
```
Comment (alon710) [inline handlers/monday.py:11]: Should also get logger

File: handlers/monday.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+    def get_boards(self):
+        return self.client.boards
```
Comment (alon710) [inline handlers/monday.py]:
```suggestion
        return self.client.boards.fetch_boards()
```

File: scraper.py [hunk 5]
```diff
@@ -29,6 +29,13 @@ def get_logger() -> logging.Logger:
     username = config["MOODLE_USERNAME"]
     password = config["MOODLE_PASSWORD"]
     base_url = config["MOODLE_BASEURL"]
+    monday_token = config["MONDAY_API_KEY"]
+    monday = MondayHandler(monday_token)
+    boards = monday.get_boards()
+    board = boards.fetch_boards()
```
Comment (alon710) [inline scraper.py]: Remove

File: handlers/monday.py [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+    def get_board(self, board_id):
+        return self.client.boards.get_board(board_id)
+
+    def add_item(self, board_id, item,moodle: MoodleHandler):
```
Comment (alon710) [inline handlers/monday.py]: Monday handler should not get MoodleHandler
If you want to have access to the files, the Moodle handler should be able to download them and save the data (perhaps as base64 string) and then you should send the file data to Monday

File: scraper.py [hunk 7]
```diff
@@ -41,6 +48,12 @@ def get_logger() -> logging.Logger:
     assigns = mh.get_assigns()
 
     for assign in assigns:
+        item_id = monday.add_item(
+            board_id=board_id,
+            item=assign,
+            moodle=mh,
+        )
```
Comment (alon710) [inline scraper.py]: You should also somehow save state in order to not update two times

File: handlers/monday.py (hunk 6 above)
Comment (Orbasker) [reply handlers/monday.py]: why not? you must have the session and you only want to temp sace the file, its not a constant types and can be to much for base64 no?

File: .env [hunk 8]
```diff
@@ -1,3 +0,0 @@
-MOODLE_USERNAME=
-MOODLE_PASSWORD=
-MOODLE_BASEURL=
```
Comment (alon710) [inline .env]: Add new variables

File: handlers/monday.py [hunk 9]
```diff
@@ -0,0 +1,54 @@
+import os
+
+from monday import MondayClient
+
+from handlers.moodle import MoodleHandler
+
+
+class MondayHandler:
+    def __init__(self, token, logger=None):
```
Comment (alon710) [inline handlers/monday.py]: Add type annotation

File: handlers/monday.py [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+        self.client = MondayClient(token)
+        self.logger = logger
+
+    def get_boards(self):
```
Comment (alon710) [inline handlers/monday.py]: Add return value type

File: handlers/monday.py [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+        return self.client.boards
+
+    def get_board(self, board_id):
+        return self.client.boards.get_board(board_id)
```
Comment (alon710) [inline handlers/monday.py]: Add type annotation and return value type

File: scraper.py [hunk 12]
```diff
@@ -29,8 +30,14 @@ def get_logger() -> logging.Logger:
     username = config["MOODLE_USERNAME"]
     password = config["MOODLE_PASSWORD"]
     base_url = config["MOODLE_BASEURL"]
+    monday_token = config["MONDAY_API_KEY"]
+    board_id = config["MONDAY_BOARD_ID"]
     logger = get_logger()
 
+    monday = MondayHandler(token=monday_token, logger=logger)
+    boards = monday.get_boards()
+    items = boards.fetch_items_by_board_id(board_id)
```
Comment (alon710) [inline scraper.py]: Change to use the monday handler function

**Pull request merged**